src/test/java/core/basesyntax/service/ActionStrategyImplTest.java

```diff
@@ -0,0 +1,86 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.action.ActionHandler;
+import core.basesyntax.service.action.BalanceAction;
+import core.basesyntax.service.action.PurchaseAction;
+import core.basesyntax.service.action.ReturnAction;
+import core.basesyntax.service.action.SupplyAction;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ActionStrategyImplTest {
+    private static final Map<Operation, ActionHandler> actionHandlerMap = Map.of(
+            Operation.BALANCE, new BalanceAction(),
+            Operation.PURCHASE, new PurchaseAction(),
+            Operation.RETURN, new ReturnAction(),
+            Operation.SUPPLY, new SupplyAction()
+    );
+    private final ActionStrategy strategy = new ActionStrategyImpl(actionHandlerMap);
+
+    @Test
+    void checkBalanceActionGet_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction(Operation.BALANCE, "banana", 20);
+        assertEquals(BalanceAction.class, strategy.get(fruitTransaction.getOperation()).getClass());
+    }
+
+    @Test
+    void checkBalanceActionGetWithWrongAmount_NotOk() {
+        FruitTransaction fruitTransaction =
+                new FruitTransaction(Operation.BALANCE, "banana", -20);
+        assertThrows(RuntimeException.class, () ->
+                strategy.get(fruitTransaction.getOperation())
+                        .count(fruitTransaction.getFruit(), fruitTransaction.getQuantity()));
+    }
```

Comment:
It’s not strategy logic. Remove negative test cases from this class

Reply:
mentor told me to add negative tests 
https://github.com/mate-academy/jv-fruit-shop-tests/pull/1079#discussion_r1855675576

Reply:
I believe what Lilia meant was that ActionStrategyImpl is not responsible for throwing an exception if the count is negative. ActionStrategyImpl should return you the appropriate handler based on the operation type, and that's it.

So, you negative test case can be, that method get() of ActionStrategyImpl will throw a RuntimeException if `actionHandlerMap` is null. Smth like this + check error message content:
```
@Test
    void getActionHandler_mapIsNull_notOk() {
        ActionStrategy actionStategy = new ActionStrategyImpl(null);
        assertThrows(RuntimeException.class, () -> {
            strategyWithNullMap.getOperationHandler(Operation.BALANCE));
    }
```